olp-cpp-sdk-dataservice-read/src/repositories/NamedMutex.cpp

```diff
@@ -30,11 +30,14 @@ class NamedMutexStorage::Impl {
  public:
   std::mutex& AquireLock(const std::string& resource);
   void ReleaseLock(const std::string& resource);
+  void SetError(const std::string& resource, const client::ApiError& error);
+  boost::optional<client::ApiError> GetError(const std::string& resource);
 
  private:
   struct RefCounterMutex {
     std::mutex mutex;
     uint32_t use_count{0u};
+    boost::optional<client::ApiError> optional_error;
```

Comment:
I do not see where this error is reset. Or we do not need to reset it?

Reply:
RefCounterMutex instance is removed when the last its user calls ReleaseLock. So we don't need to clear the error explicitly